src/photographers/test/photographers.component.spec.ts

```diff
@@ -1,74 +1,222 @@
+  });
+
+  it('should unsubscribe on destroy', () => {
+    spyOn(component.keyUpSubscribe, 'unsubscribe');
+    spyOn(component.getTranslationSubscribe, 'unsubscribe');
+    spyOn(component.photographersServiceSubscribe, 'unsubscribe');
+
+    component.ngOnDestroy();
+
+    expect(component.keyUpSubscribe.unsubscribe).toHaveBeenCalled();
+    expect(component.getTranslationSubscribe.unsubscribe).toHaveBeenCalled();
+    expect(component.photographersServiceSubscribe.unsubscribe).toHaveBeenCalled();
+  });
+
+  xit('should blur search input on mobile', () => {
+    // TODO how to test this?
+    component.isDesktop = false;
+  });
+
+  it('toggleLeftSide should toggle sidebar', () => {
+    spyOn(component, 'toggleLeftSide');
+    component.ngAfterViewInit();
+
+    component.isDesktop = false;
+    const toggleBtn = fixture.debugElement.query(By.css('.sort-country'))
+      .nativeElement
+      .click();
+
+    expect(component.toggleLeftSide).toHaveBeenCalled();
+  });
+
+  describe('PhotographerComponent: photographers card', () => {
+    beforeEach(() => {
+      spyOn(photographersService, 'getPhotographers').and.returnValue(Observable.of({
+        data: {
+          countryList: countryListData,
+          photographersList: photographerListData
+        }
+      }));
+      spyOn(router, 'navigate').and.stub();
+
+      component.ngAfterViewInit();
+      fixture.detectChanges();
+    });
+
+    it('should create cards', () => {
+      const cards = fixture.debugElement.queryAll(By.css('.photographer-card'));
+
+      expect(cards.length).toEqual(photographerListData.length);
+    });
+
+    it('should create avatar', () => {
+      const cardAvatar = fixture.debugElement.query(By.css('.photographer-portrait'));
+
+      expect(cardAvatar.styles['background-image']).toEqual(photographerListData[0].avatar);
+    });
+
+    it('should create card name', () => {
+      const cardName = fixture.debugElement.query(By.css('h3')).nativeElement;
+
+      expect(cardName.textContent).toEqual(photographerListData[0].name);
+    });
+
+    it('show number of photographer images', () => {
+      const imagesCounter = fixture.debugElement.query(By.css('.photographer-material span')).nativeElement;
+
+      expect(Number(imagesCounter.textContent)).toEqual(photographerListData[0].images);
+    });
+
+    it('show number of photographer videos', () => {
+      const videosCounter = fixture.debugElement.queryAll(By.css('.photographer-material span'))[1].nativeElement;
+
+      expect(Number(videosCounter.textContent)).toEqual(photographerListData[0].video);
+    });
+
+    it('show number of photographer places', () => {
+      const placesCounter = fixture.debugElement.queryAll(By.css('.photographer-material span'))[2].nativeElement;
+
+      expect(Number(placesCounter.textContent)).toEqual(photographerListData[0].places);
+    });
+
+    xit('should navigate to photographer page on click', () => {
```

Comment:
where is reson for xiting?

Reply:
Use `// fixme: description...`

Reply:
TODO with explanations is on the next line